enhancements/multi-arch/dynamic-imagestream-importmode-setting.md

```diff
@@ -0,0 +1,390 @@
+of the cluster. This means that if the OCP release
+payload is a singlemanifest payload, the
+importMode for all imagestreams by default will be
+`Legacy` and if the payload is a multi payload the
+default will be `PreserveOriginal`. 
+
+## Motivation
+
+This change ensures that imagestreams will be
+compatible for single and multi arch compute
+clusters without needing manual intervention.
+Users who install or upgrade to the multi
+release payload
+(https://mirror.openshift.com/pub/openshift-v4/multi/)
+and want to add compute nodes of differing
+architectures, need not change the behaviour
+of any newly imported imagestreams or any scripts
+which import them. For a cluster installed with
+single arch payload, imagestreams default import
+mode would always be set to `Legacy` because it
+is sufficient if the single image manifest is
+imported and for a cluster installed with the
+multi payload it would be `PreserveOriginal`
+which would import the manifestlist in anticipation
+that users would add/have already added nodes
+of other architectures (which do not match the
+control plane's).
+
+### User Stories
+
+- As an Openshift user, I do not want to manually
+  change the importMode of imagestreams I want to
+  import on the cluster if I add/remove nodes of
+  a different architecture to it.
+- As an Openshift user, I do not want to change my
+  existing scripts and oc commands to add extra
+  flags to enable manifestlist imports on
+  imagestreams
+- As an Openshift user, I want my imagestreams to
+  work with workloads deployed on any node
+  irrespective of whether I deploy on a
+  single or a multi-arch compute cluster.
+- As an Openshift user, I want to be able to toggle
+  importMode globally for all imagestreams through
+  configuration.
+
+### Goals
+
+- Set the imagestream's import mode based on
+  payload architecture (single vs multi) for newly
+  created imagestreams.
+- Dynamically alter import mode type on upgrade
+  from a single arch to a multi paylaod and vice
+  versa.
+- Allow users to control the import mode setting
+  manually through the `image.config.openshift.io`
+  image config CRD.
+
+### Non-Goals
+
+- Describe importmode and what it accomplishes
+  relating to imagestreams
+- Existing imagestreams in the cluster will not be
+  altered
+- No other imagestream fields are being considered
+  for such a change
+- Sparse manifest imports is not being considered
+
+## Proposal
+
+The proposal is for CVO to expose the
+architecture/type of payload installed on the
+cluster and for apiserver to use that information
+to set the importMode for imagestreams globally.
+
+### Workflow Description
+ 
+#### Installation
+1. User triggers installation through
+openshift-installer
+2. CVO inspects the payload and updates the status
+field of `ClusterVersion` with the payload
+architecture/type (inferred through
+`release.openshift.io/architecture`)
+3. openshift-apiserver operator looks to see if
```

Comment:
openshift-apiserver-operator please.  Have that set a configuration value on the openshift-apiserver so that the openshift-apiserver isn't dynamically changing values.

Reply:
fixed.